### PR TITLE
Fix broken markdown in actions' details

### PR DESCRIPTION
**Pull request by revolter**
### This is how the docs would look like if they would get updated using the lastest version:

screen shot 2018-04-20 at 23 06 54

### And how it looks like in the present and also after this PR:

screen shot 2018-04-20 at 23 07 11

---

### Other differences between the live version and the new version combined from #12287 and this one:

Before:

screen shot 2018-04-20 at 23 05 06

After:

screen shot 2018-04-20 at 23 05 14

Before:

screen shot 2018-04-16 at 20 56 03

After:

screen shot 2018-04-16 at 20 56 11

Before:

screen shot 2018-04-16 at 20 50 31

After:

screen shot 2018-04-16 at 20 51 17

Before:

screen shot 2018-04-15 at 20 15 10

After:

screen shot 2018-04-15 at 20 15 17

Before:

screen shot 2018-04-11 at 23 47 55

After:

screen shot 2018-04-20 at 23 12 24

Before:

screen shot 2018-04-20 at 23 17 12

After:

screen shot 2018-04-20 at 23 17 28

### CODE REVIEW

**Comment by joshdholtz** [comment]
@revolter I ran `fastlane action add` and got...
```sh
➜  fastlane git:(revolter-hotfix/deprecation-details-fixes) fl action adb
[✔] 🚀
bundler: failed to load command: fastlane (/Users/josh/.gem/ruby/2.2.10/bin/fastlane)
SyntaxError: /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/get_push_certificate.rb:33: syntax error, unexpected <<
        sample = <<~SAMPLE.markdown_sample
                   ^
/Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/get_push_certificate.rb:34: syntax error, unexpected tXSTRING_BEG, expecting keyword_end
          ```ruby
             ^
/Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/get_push_certificate.rb:40: syntax error, unexpected tXSTRING_BEG, expecting keyword_end
          ```
            ^
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/actions_helper.rb:94:in `require'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/actions_helper.rb:94:in `block in load_default_actions'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/actions_helper.rb:93:in `each'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/actions/actions_helper.rb:93:in `load_default_actions'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane.rb:37:in `load_actions'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/commands_generator.rb:34:in `start'
  /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
  /Users/josh/Projects/fastlane/fastlane/bin/fastlane:20:in `<top (required)>'
  /Users/josh/.gem/ruby/2.2.10/bin/fastlane:23:in `load'
  /Users/josh/.gem/ruby/2.2.10/bin/fastlane:23:in `<top (required)>'
```

**Comment by revolter** [comment]
Oh, right, I'm stupid. I'll see if I can fix it now.

**Review by joshdholtz — CHANGES_REQUESTED** [review]
All of the `<<~` need to be replaced with `<<-` 😊

**Comment by revolter** [comment]
@joshdholtz, Fixed it. But I rebased it because I had a couple of useless commits and I didn't want to revert them.

**Comment by joshdholtz** [comment]
@revolter Is this spacing too large for the lists when viewing in CLI? 🤔 
```sh
+----------------------------------------------------------------------------------------------------------------------------+
|                                                        add_git_tag                                                         |
+----------------------------------------------------------------------------------------------------------------------------+
| This will add an annotated git tag to the current branch                                                                   |
|                                                                                                                            |
| This will automatically tag your build with the following format: `<grouping>/<lane>/<prefix><build_number>`, where:       |
|                                                                                                                            |
| -           `grouping` is just to keep your tags organised under one 'folder', defaults to 'builds'                        |
| -           `lane` is the name of the current fastlane lane                                                                |
| -           `prefix` is anything you want to stick in front of the version number, e.g. 'v'                                |
| -           `postfix` is anything you want to stick at the end of the version number, e.g. '-RC1'                          |
| -           `build_number` is the build number, which defaults to the value emitted by the `increment_build_number` action |
|                                                                                                                            |
| For example, for build 1234 in the 'appstore' lane, it will tag the commit with `builds/appstore/1234`.                    |
|                                                                                                                            |
| Created by lmirosevic, maschall                                                                                            |
+----------------------------------------------------------------------------------------------------------------------------+
```

**Comment by revolter** [comment]
@joshdholtz, Nope. But now, if the link text is the same as the URL, I remove one of them and add an emoji:

```
+----------------------------------------------------------------------------------------------------------------------------------+
|                                                              sonar                                                               |
+----------------------------------------------------------------------------------------------------------------------------------+
| Invokes sonar-scanner to programmatically run SonarQube analysis                                                                 |
|                                                                                                                                  |
| See http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner 🔗 for details.                                      |
| It can process unit test results if formatted as junit report as shown in [xctest](https://docs.fastlane.tools/actions/xctest/)  |
| action. It can also integrate coverage reports in Cobertura format, which can be transformed into by the                         |
| [slather](https://docs.fastlane.tools/actions/slather/) action.                                                                  |
|                                                                                                                                  |
| Created by c_gretzki                                                                                                             |
+----------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by revolter** [comment]
I tried using `tty-markdown` but it doesn't work with our tables if a markdown elements needs to show up on 2 lines.

File: fastlane/lib/fastlane/actions/create_app_online.rb [hunk 1]
```diff
@@ -25,13 +25,12 @@ def self.description
         "Creates the given application on iTC and the Dev Portal (via _produce_)"
       end
 
-      def details
+      def self.details
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/create_app_online.rb]: Good find!

**Comment by joshdholtz** [comment]
@revolter Do we want the tickticktick and `no highlight` showing like it does in CLI here? 🤔 

```sh
+-------------------------------------------------------------------------------------------------------+
|                                          get_github_release                                           |
+-------------------------------------------------------------------------------------------------------+
| This will verify if a given release version is available on GitHub                                    |
|                                                                                                       |
| This will return all information about a release. For example:                                        |
|                                                                                                       |
| ```no-highlight                                                                                       |
| {                                                                                                     |
|   "url"=>"https://api.github.com/repos/KrauseFx/fastlane/releases/1537713",                           |
|    "assets_url"=>"https://api.github.com/repos/KrauseFx/fastlane/releases/1537713/assets",            |
|    "upload_url"=>"https://uploads.github.com/repos/KrauseFx/fastlane/releases/1537713/assets{?name}", |
|    "html_url"=>"https://github.com/fastlane/fastlane/releases/tag/1.8.0",                             |
|    "id"=>1537713,                                                                                     |
|    "tag_name"=>"1.8.0",                                                                               |
|    "target_commitish"=>"master",                                                                      |
|    "name"=>"1.8.0 Switch Lanes & Pass Parameters",                                                    |
|    "draft"=>false,                                                                                    |
|    "author"=>                                                                                         |
|     {"login"=>"KrauseFx",                                                                             |
|      "id"=>869950,                                                                                    |
|      "avatar_url"=>"https://avatars.githubusercontent.com/u/869950?v=3",                              |
|      "gravatar_id"=>"",                                                                               |
|      "url"=>"https://api.github.com/users/KrauseFx",                                                  |
|      "html_url"=>"https://github.com/fastlane",                                                       |
|      "followers_url"=>"https://api.github.com/users/KrauseFx/followers",                              |
|      "following_url"=>"https://api.github.com/users/KrauseFx/following{/other_user}",                 |
|      "gists_url"=>"https://api.github.com/users/KrauseFx/gists{/gist_id}",                            |
|      "starred_url"=>"https://api.github.com/users/KrauseFx/starred{/owner}{/repo}",                   |
|      "subscriptions_url"=>"https://api.github.com/users/KrauseFx/subscriptions",                      |
|      "organizations_url"=>"https://api.github.com/users/KrauseFx/orgs",                               |
|      "repos_url"=>"https://api.github.com/users/KrauseFx/repos",                                      |
|      "events_url"=>"https://api.github.com/users/KrauseFx/events{/privacy}",                          |
|      "received_events_url"=>"https://api.github.com/users/KrauseFx/received_events",                  |
|      "type"=>"User",                                                                                  |
|      "site_admin"=>false},                                                                            |
|    "prerelease"=>false,                                                                               |
|    "created_at"=>"2015-07-14T23:33:01Z",                                                              |
|    "published_at"=>"2015-07-14T23:44:10Z",                                                            |
|    "assets"=>[],                                                                                      |
|    "tarball_url"=>"https://api.github.com/repos/KrauseFx/fastlane/tarball/1.8.0",                     |
|    "zipball_url"=>"https://api.github.com/repos/KrauseFx/fastlane/zipball/1.8.0",                     |
|    "body"=> ...Markdown...                                                                            |
|   "This is one of the biggest updates of _fastlane_ yet"                                              |
| }                                                                                                     |
| ```                                                                                                   |
|                                                                                                       |
| Created by KrauseFx, czechboy0, jaleksynas, tommeier                                                  |
+-------------------------------------------------------------------------------------------------------+
```

**Comment by revolter** [comment]
Well, one option would be to remove them, and I think that it will be a little bit harder to quickly scan the text and the different parts, so I'll leave them (some actions already had these too, so it's not new) as some kind of separators.

Another option would be to highlight them like a Markdown parser but that's complicated.

**Comment by joshdholtz** [comment]
@revolter Ahhh yeah yeah, I _did_ notice that some of the actions previously used the markdown link syntax.

I'm not sure how clean this would be but could we use a regex like this (https://stackoverflow.com/a/31192012) to extract any markdown links and just replace it with the url?

**Comment by revolter** [comment]
Let's take `sonar` for example.

Its details are:

```
See [http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner](http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner) for details.
It can process unit test results if formatted as junit report as shown in [xctest](https://docs.fastlane.tools/actions/xctest/) action. It can also integrate coverage reports in Cobertura format, which can be transformed into by the [slather](https://docs.fastlane.tools/actions/slather/) action.
```

In terminal, it shows like this:

```
See http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner 🔗 for details.
It can process unit test results if formatted as junit report as shown in [xctest](https://docs.fastlane.tools/actions/xctest/) action. It can also integrate coverage reports in Cobertura format, which can be transformed into by the [slather](https://docs.fastlane.tools/actions/slather/) action.
```

Are you saying to change the second like to:

```
It can process unit test results if formatted as junit report as shown in https://docs.fastlane.tools/actions/xctest/ 🔗 action. It can also integrate coverage reports in Cobertura format, which can be transformed into by the https://docs.fastlane.tools/actions/slather/ 🔗 action.
```

?

_(the `🔗` is optional, I just added it for Markdown links where the text was the same, in which case I remove it)_

This would make incorrect phrasing for this `appetize`'s details line:

```
To integrate appetize into your GitHub workflow check out the [device_grid guide](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/device_grid/README.md).
```

where we want the text **and** the link too.

What if we get rid of the not user friendly Markdown syntax by replacing the links, in these example, like this:

```
See http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner 🔗 for details.
It can process unit test results if formatted as junit report as shown in "xctest" (https://docs.fastlane.tools/actions/xctest/ 🔗) action. It can also integrate coverage reports in Cobertura format, which can be transformed into by the "slather" (https://docs.fastlane.tools/actions/slather/ 🔗) action.
```

and

```
To integrate appetize into your GitHub workflow check out the "device_grid guide" (https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/device_grid/README.md 🔗).
```

**Comment by joshdholtz** [comment]
@revolter I really like your suggestion with the `"link name" (link text)` 👌 That is actually much easier to read, IMO, than the other ways on CLI. Is that too much work to do? 😊

**Comment by joshdholtz** [comment]
@revolter I'm always down for emojis 🙃

**Comment by joshdholtz** [comment]
@revolter That markdown link change is so good ✊

**Review by joshdholtz — APPROVED** [review]
I'm good with merging this if you are good with merging this!

**Pull request merged**